src/adapter/src/coord/sequencer.rs

```diff
@@ -283,6 +285,28 @@ impl Coordinator {
                 | Plan::AbortTransaction(AbortTransactionPlan {
                     ref transaction_type,
                 }) => {
+                    // Serialize DDL transactions.
+                    if ctx.session().transaction().is_ddl() {
```

Comment:
Do we need to further check that `must_serialize_ddl()` is true?

Reply:
No, because that only checks Statements not Plans. However I've added doc strings scaring people into making sure that users of the DDL mode always return false in must_serialize_ddl(). I hope that's sufficient for now.